Continue cleaning up compiler configurations

Constellation opened the pull request:
#### 924586597e706124d512961ccd9083a8736ad985
<pre>
Continue cleaning up compiler configurations
<a href="https://bugs.webkit.org/show_bug.cgi?id=274438">https://bugs.webkit.org/show_bug.cgi?id=274438</a>
<a href="https://rdar.apple.com/128441724">rdar://128441724</a>

Reviewed by Ross Kirsling.

This patch further simplifies compiler configurations.

1. Drop MINGW, nobody is supporting it.
2. Remove MSVC hacks further.
3. Remove most of COMPILER(GCC_COMPATIBLE) in JavaScriptCore.
4. Fix CodeBlock&apos;s static_assert, which was not effective before.

* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/Sources.txt:
* Source/JavaScriptCore/assembler/ARM64Assembler.h:
* Source/JavaScriptCore/assembler/MacroAssemblerARM64.cpp:
* Source/JavaScriptCore/assembler/MacroAssemblerARMv7.cpp:
* Source/JavaScriptCore/bytecode/CodeBlock.h:
* Source/JavaScriptCore/heap/GCMemoryOperations.h:
(JSC::gcSafeMemcpy):
(JSC::gcSafeMemmove):
(JSC::gcSafeZeroMemory):
* Source/JavaScriptCore/jit/CCallHelpers.h:
(JSC::CCallHelpers::marshallArgumentRegister):
* Source/JavaScriptCore/jit/JITOperations.cpp:
(JSC::JSC_DEFINE_NOEXCEPT_JIT_OPERATION):
* Source/JavaScriptCore/runtime/ArrayConventions.cpp:
(JSC::clearArrayMemset):
* Source/JavaScriptCore/runtime/EnsureStillAliveHere.cpp: Removed.
* Source/JavaScriptCore/runtime/EnsureStillAliveHere.h:
* Source/JavaScriptCore/runtime/ExceptionScope.h:
* Source/JavaScriptCore/runtime/ISO8601.h:
* Source/JavaScriptCore/runtime/JSBigInt.cpp:
(JSC::JSBigInt::digitDiv):
* Source/WTF/wtf/Assertions.h:
* Source/WTF/wtf/Compiler.h:
* Source/WTF/wtf/CurrentTime.cpp:
(WTF::highResUpTime):
* Source/WTF/wtf/InlineASM.h:
* Source/WTF/wtf/win/ThreadingWin.cpp:
(WTF::Thread::initializeCurrentThreadInternal):
* Source/WebCore/platform/graphics/FormatConverter.cpp:
(WebCore::FormatConverter::convert):

Canonical link: <a href="https://commits.webkit.org/279051@main">https://commits.webkit.org/279051@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/487176507114ed940e8472b319b04fe692beb979

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/52343 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/31675 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/4764 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/55617 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/3066 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/54648 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/38116 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/2765 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/42572 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/1957 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/54439 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/29310 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45174 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/23652 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/26534 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/2449 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1225 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/45692 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/48434 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/2599 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57213 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/51851 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/27469 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/2632 "Passed tests") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/28702 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45294 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/29614 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/64158 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7672 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/28447 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12145 "Found 9 new JSC stress test failures: stress/map-iteration.js.dfg-eager, stress/proxy-set-prototype-of.js.dfg-eager, stress/spread-non-array.js.no-llint, stress/webidl-tokenizer-for-of.js.bytecode-cache, wasm.yaml/wasm/function-tests/add-12.js.wasm-eager, wasm.yaml/wasm/function-tests/br-if-as-return.js.wasm-eager, wasm.yaml/wasm/js-api/Module-compile.js.wasm-eager, wasm.yaml/wasm/stress/osr-entry-many-stacks-i32.js.wasm-eager, wasm.yaml/wasm/stress/too-many-return-types.js.wasm-eager (failure)") | 
<!--EWS-Status-Bubble-End-->

rkirsling commented on Source/WTF/wtf/Compiler.h:
```diff
@@ -468,12 +382,8 @@
-
 #if !defined(UNUSED_VARIADIC_PARAMS)
-#define UNUSED_VARIADIC_PARAMS
+#define UNUSED_VARIADIC_PARAMS __attribute__((unused))
```
Looks like the last one to remove from the previous patch is the
```
#if !defined(WARN_UNUSED_RETURN)
#define WARN_UNUSED_RETURN
#endif
```
right below here.

Constellation replied:
Done

rkirsling commented on Source/JavaScriptCore/bytecode/CodeBlock.h:
```diff
@@ -992,8 +992,8 @@ class CodeBlock : public JSCell {
-#if defined(NDEBUG) && !defined(ASSERT_ENABLED) && COMPILER(GCC_COMPATIBLE)
-static_assert(sizeof(CodeBlock) <= 224, "Keep it small for memory saving");
+#if !ASSERT_ENABLED
+static_assert(sizeof(CodeBlock) <= 232, "Keep it small for memory saving");
```
It would be good to describe this value change in the changelog.

Constellation replied:
Done.

rkirsling commented on Source/JavaScriptCore/heap/GCMemoryOperations.h:
```diff
@@ -251,7 +251,7 @@ ALWAYS_INLINE void gcSafeZeroMemory(T* dst, size_t bytes)
     RELEASE_ASSERT(bytes % 8 == 0);
 #if USE(JSVALUE64)
-#if COMPILER(GCC_COMPATIBLE) && (CPU(X86_64) || CPU(ARM64))
+#if CPU(X86_64) || CPU(ARM64)
```
I think we can actually remove this outer `#if`, so that there's just `#if CPU(X86_64)`, `#elif CPU(ARM64)`, and `#else`.

Constellation replied:
Done

rkirsling reviewed (approved):
r=me with one more comment

Constellation committed:
Continue cleaning up compiler configurations
https://bugs.webkit.org/show_bug.cgi?id=274438
rdar://128441724

Reviewed by Ross Kirsling.

This patch further simplifies compiler configurations.

1. Drop MINGW, nobody is supporting it.
2. Remove MSVC hacks further.
3. Remove most of COMPILER(GCC_COMPATIBLE) in JavaScriptCore.
4. Fix CodeBlock's static_assert, which was not effective before.

* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/Sources.txt:
* Source/JavaScriptCore/assembler/ARM64Assembler.h:
* Source/JavaScriptCore/assembler/MacroAssemblerARM64.cpp:
* Source/JavaScriptCore/assembler/MacroAssemblerARMv7.cpp:
* Source/JavaScriptCore/bytecode/CodeBlock.h:
* Source/JavaScriptCore/heap/GCMemoryOperations.h:
(JSC::gcSafeMemcpy):
(JSC::gcSafeMemmove):
(JSC::gcSafeZeroMemory):
* Source/JavaScriptCore/jit/CCallHelpers.h:
(JSC::CCallHelpers::marshallArgumentRegister):
* Source/JavaScriptCore/jit/JITOperations.cpp:
(JSC::JSC_DEFINE_NOEXCEPT_JIT_OPERATION):
* Source/JavaScriptCore/runtime/ArrayConventions.cpp:
(JSC::clearArrayMemset):
* Source/JavaScriptCore/runtime/EnsureStillAliveHere.cpp: Removed.
* Source/JavaScriptCore/runtime/EnsureStillAliveHere.h:
* Source/JavaScriptCore/runtime/ExceptionScope.h:
* Source/JavaScriptCore/runtime/ISO8601.h:
* Source/JavaScriptCore/runtime/JSBigInt.cpp:
(JSC::JSBigInt::digitDiv):
* Source/WTF/wtf/Assertions.h:
* Source/WTF/wtf/Compiler.h:
* Source/WTF/wtf/CurrentTime.cpp:
(WTF::highResUpTime):
* Source/WTF/wtf/InlineASM.h:
* Source/WTF/wtf/win/ThreadingWin.cpp:
(WTF::Thread::initializeCurrentThreadInternal):
* Source/WebCore/platform/graphics/FormatConverter.cpp:
(WebCore::FormatConverter::convert):

Canonical link: https://commits.webkit.org/279051@main
